src/queue.js

```diff
@@ -0,0 +1,47 @@
+export class Node {
+  constructor(value) {
+    this.data = value
+    this.next = null
+  }
+}
+
+export default class Queue {
+  constructor() {
+    this.size = 0
+    this.head = null
+    this.tail = null
+  }
+
+  enqueue = (value) => {
+    const node = new Node(value)
+      if (this.size) {
```

Comment:
Indentation is off.  Also, I recommend the use of explicit checks in JavaScript:
```js
if( this.size !== 0 ) {
```